usb: Cleanup for multiplied defines

eobalski opened the pull request:
Some of defines are present in several header files.
Those defines are the same with value but with different naming.

Common defines are brought to usb_common.h

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

Commit:
usb: Cleanup for multiplied defines

Some of defines are present in several header files.
Those defines are the same with value but with different naming.

Common defines are brought to usb_common.h

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

jfischer-phytec-iot commented on include/usb/usb_common.h:
```diff
@@ -65,10 +65,14 @@
 #define USB_DEVICE_CAPABILITY_DESC	0x10
 #define USB_HID_DESC			0x21
 #define USB_HID_REPORT_DESC		0x22
+#define CS_INTERFACE			0x24
```
Please rename to USB_CS_INTERFACE_DESC and USB_CS_ENDPOINT_DESC

emob-nordic replied:
done

Commit:
usb: Rename defines to match present naming convention

This commit renames two defines:
CS_INTERFACE -> USB_CS_INTERFACE_DESC
CS_ENDPOINT -> USB_CS_ENDPOINT_DESC
in order to match current naming convention when it comes
to descriptors fields.

All relevant files are updated to match renamed macros.

Signed-off-by: Emil Obalski <emil.obalski@nordicsemi.no>

finikorg commented on subsys/usb/class/cdc_acm.c:
```diff
@@ -118,15 +118,15 @@ struct usb_cdc_acm_config {
 #define INITIALIZER_IF_HDR						\
```
I think we also can move initializers to the common header.

emob-nordic replied:
For some yes, but there exist a group of class specific ones which better stay here. We can consider this change later and decide they best solution.
